automation/get_target_servers.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/local/bin/python
+"""Script to get target server from Apigee"""
+
+import argparse
+import zipfile
+import io
+from requests import Response
+from requests import Session
+
+from service import apigee_auth
+
+# Global session used for all requests.
+REQUEST = Session()
+APIGEE_API_URL = 'https://api.enterprise.apigee.com/v1/organizations/{}/environments/{}'
+
+
+class TargetServer:
+    def __init__(self, organization: str, environment: str, target_server: any):
+        self.organization = organization
+        self.environment = environment
+        self.target_server = target_server
+
+    def __str__(self):
+        return f"TargetServer: organization: {self.organization}, environment: {self.environment}, target_server: {self.target_server}"
+
+
+def get_target_server(org_name: str, env_name: str, target_server_name: str):
+    """Retrieves the target server by name of the given organization name and environment name"""
+
+    response = REQUEST.get((APIGEE_API_URL + '/targetservers/{}').format(org_name, env_name, target_server_name))
+
+    if response.status_code != 200:
+        raise Exception(print_error(response))
+
+    target_server = response.json()
+
+    return target_server
+
+
+def exists_env_in_org(org_name: str, env_name: str):
+    """Checks if environment exists in organization"""
+
+    response = REQUEST.get(APIGEE_API_URL.format(org_name, env_name))
+
+    return response.status_code == 200
+
+
+def print_error(response: Response) -> str:
+    """Prints the error returned from an API call"""
+    return f'Error: {response.status_code} - {response.reason}. \n {response.text}'
+
+
+def parse_args():
+    """Defines which arguments are needed for the script to run."""
+    parser = argparse.ArgumentParser(
+        description='Gets target server by name from Apigee')
+    req_grp = parser.add_argument_group(title='required arguments')
+    req_grp.add_argument(
+        '-tsn',
+        '--target_server_name',
+        help='name of the target server',
+        required=True)
+    req_grp.add_argument(
+        '-env',
+        '--environments',
+        help='names of the environments',
+        required=True)
+    req_grp.add_argument(
```

Comment:
As discussed maybe it is not the best idea to accept multiple organizations, although I understand why you did it.  Is it possible to only accept one org? and an array of environments